File: demos-news/MemoryManagementDemos/MemoryManagementDemos/Superhero.m
//
//  Superhero.m
//  MemoryManagementDemos
//
//  Created by Doncho Minkov on 1/26/16.
//  Copyright © 2016 Doncho Minkov. All rights reserved.
//

#import "Superhero.h"

@implementation Superhero

-(instancetype)initWithName:(NSString *)name {
    self = [super init];
    if(self) {
        self.name = name;
    }
    return self;
}


-(Superhero *)superheroWithName:(NSString *)name    {
    return [[Superhero alloc] initWithName:name];
}

-(void)dealloc {
    NSLog(@"IN the dealloc");
}

@end

File: demos/SQLiteExplore/SQLiteExplore/Superhero.swift
//
//  Superhero.swift
//  SQLiteExplore
//
//  Created by Doncho Minkov on 3/27/17.
//  Copyright © 2017 Doncho Minkov. All rights reserved.
//

import Foundation

class Superhero {
    var id: Int64
    var name: String?
    var secretIdentity: String?
    
    convenience init(withName name: String, andSecretIdentity secretIdentity: String) {
        let id =  Int64(arc4random())
        self.init(withId: id, name: name, andSecretIdentity: secretIdentity)
    }
    
    init(withId id: Int64, name: String, andSecretIdentity secretIdentity: String){
        self.id = id
        self.name = name
        self.secretIdentity = secretIdentity
    }
}
